File: problem-solving/baekjoon/11726_2xn_타일링.go
// https://www.acmicpc.net/problem/11726
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)

	var count = make([]int, n+1)
	for i := 1; i < n+1; i++ {
		if i < 3 {
			count[i] = i
		} else {
			count[i] = (count[i-1] + count[i-2]) % 10007
		}
	}
	fmt.Fprintln(writer, count[n])
}

File: problem-solving/baekjoon/11726_2xn_타일링.py
# https://www.acmicpc.net/problem/11726
import sys

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    count = [0 for i in range(n+1)]
    for i in range(1, n+1):
        if i < 3:
            count[i] = i
        else:
            count[i] = (count[i-1] + count[i-2]) % 10007
    print(count[n])